great_expectations_cloud/agent/agent.py

```diff
@@ -210,13 +210,16 @@ def _handle_event_as_thread_enter(self, event_context: EventContext) -> None:
             self._redeliver_msg_task = loop.create_task(event_context.redeliver_message())
             return
 
+        data_context = self.get_data_context(event_context=event_context)
         # ensure that great_expectations.http requests to GX Cloud include the job_id/correlation_id
-        self._set_http_session_headers(correlation_id=event_context.correlation_id)
+        self._set_http_session_headers(
```

Comment:
This is the key change : when the Agent enters a job thread, it runs `get_data_context()` to return either `self._data_context` in the agent, or the job-DataContext in the GX-runner.  

The  returned DataContext is then used to `_set_http_session_headers()` for the current job.

Reply:
You'll also want to be able to change the `HeaderName.USER_AGENT` value on line 472/402 so that we can distinguish between Agent calls and runner class (and the versions used).

Reply:
That's a great point @Kilo59... with the dependency on the values in `constants.py` and since we are generating a new log message and header, it might be worth considering just overriding this method in the `gx-runner`. Can we discuss @anthonyburdi ?

Reply:
@Kilo59 We ended up using overridable methods to get `HeaderName` and `USER_AGENT_HEADER` so that in the runner we can pull from a new constants.py file, see: https://github.com/greatexpectationslabs/gx-runner/pull/74
We left the header name `Agent-Job-Id` to not mess with downstream processing (mercury). If that is not a concern, happy to revisit.